### PR TITLE
Introduce Offset Gallery block

### PR DESCRIPTION
**Pull request by AnthonyLedesma**
Offset Gallery block displays images in a dynamic grid format and has no limit on images. Allows standard gallery controls and has lightbox.

**Editor**
offsetGalleryIntroduction

**Published**
offsetGalleryIntroductionPublished

Note that this block is affected by Bug reported here: https://github.com/godaddy/coblocks/issues/1143

### CODE REVIEW

**Comment by richtabor** [comment]
We'll need to add the relevant tweaks that we need to the existing gallery blocks in #1142 (if they're not already present).

**Comment by AnthonyLedesma** [comment]
Using patterns defined in #1142. Ready for review.

File: src/components/block-gallery/styles/style/_captions.scss [hunk 1]
```diff
@@ -1,5 +1,5 @@
 .has-caption-style-light {
-	.coblocks-gallery--item .coblocks-gallery--figure figcaption {
```
Comment (AnthonyLedesma) [inline src/components/block-gallery/styles/style/_captions.scss]: Removed class `.coblocks-gallery--figure` specification for support of Offset block. No visual regressions detected.

**Comment by jrtashjian** [comment]
There is a weird behavior with breakpoints right at a width of 959px, 831px, and 832px.

Kapture 2019-11-27 at 14 56 10

**Comment by richtabor** [comment]
I think that’s just because two images at those widths can’t display next to each other at those particular widths. Can you try with different images of different sizes?

**Comment by AnthonyLedesma** [comment]
@richtabor @jrtashjian 
I have different behavior with different images.

offsetResponsive

**Comment by richtabor** [comment]
That's what I thought would be the case. Different images will produce different marks where images will move onto the next "row".

**Pull request merged**